modules/schema/src/test/java/org/apache/ignite/internal/schema/marshaller/KvMarshallerTest.java

```diff
@@ -161,7 +161,14 @@ public void pojoWithFieldsOfAllTypes(MarshallerFactory factory) throws Marshalle
     @ParameterizedTest
     @MethodSource("marshallerFactoryProvider")
     public void narrowType(MarshallerFactory factory) throws MarshallerException {
-        Column[] cols = columnsAllTypes();
+        Column[] cols = new Column[]{
+                new Column("primitiveIntCol", INT32, false),
+                new Column("primitiveLongCol", INT64, false),
+                new Column("primitiveFloatCol", FLOAT, false),
+                new Column("primitiveDoubleCol", DOUBLE, false),
+                new Column("stringCol", STRING, false),
+                new Column("uuidCol", UUID, false),
+        };
         
         SchemaDescriptor schema = new SchemaDescriptor(1, cols, cols);
```

Comment:
Actually, this case is invalid.

Internally, we always know if column belong to key or value. But key and value columns with the same name can't mapped to a Record class.
So, we must fix and forbid this use case.